### PR TITLE
WHO 5 cards

### PR DESCRIPTION
**Pull request by Simisays**

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/fugitive_of_the_judoon.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Name:Fugitive of the Judoon
+ManaCost:4 G
+Types:Enchantment Saga
+K:Saga:3:DBToken,DBInvestigate,DBSearch
+SVar:DBToken:DB$ Token | TokenScript$ w_1_1_human_ward,w_4_4_alien_rhino | SpellDescription$ Create a 1/1 white Human creature token with ward {2} and a 4/4 white Alien Rhino creature token.
+SVar:DBInvestigate:DB$ Investigate | SpellDescription$ Investigate.
+SVar:DBSearch:AB$ ChangeZone | Cost$ Exile<1/Human> Exile<1/Artifact> | Origin$ Library | Destination$ Battlefield | ChangeType$ Card.Doctor | ChangeNum$ 1 | SpellDescription$You may exile a Human you control and an artifact you control. If you do, search your library for a Doctor card, put it onto the battlefield, then shuffle.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/fugitive_of_the_judoon.txt:7]: @Northmoc 
we need to check if we can make this work without the CostDesc getting in the way

File: forge-gui/res/cardsfolder/upcoming/river_songs_diary.txt [hunk 2]
```diff
@@ -0,0 +1,17 @@
+Name:River Song's Diary
+ManaCost:3
+Types:Artifact
+T:Mode$ SpellCast | ValidCard$ Instant,Sorcery | Execute$ TrigEffect | ImprintCards$ TriggeredCard | TriggerZones$ Battlefield | TriggerDescription$ Imprint — Whenever a player casts an instant or sorcery spell from their hand, exile it instead of putting it into a graveyard as it resolves.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/river_songs_diary.txt:4]: remove bad imprint param

File: forge-gui/res/cardsfolder/upcoming/river_songs_diary.txt [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+SVar:TrigEffect:DB$ Effect | ReplacementEffects$ ReMoved | RememberObjects$ TriggeredCard | SubAbility$ DBImprint
+SVar:DBImprint:DB$ Pump | ImprintCards$ TriggeredCard
+SVar:ReMoved:Event$ Moved | ValidCard$ Card.IsRemembered | Origin$ Stack | Destination$ Graveyard | Fizzle$ False | ReplaceWith$ DBExile | Description$ Exile it instead of putting it into a graveyard as it resolves.
+SVar:DBExile:DB$ ChangeZone | Defined$ ReplacedCard | Origin$ Stack | Destination$ Exile
+T:Mode$ Phase | Phase$ Upkeep | ValidPlayer$ You | CheckSVar$ X | SVarCompare$ GE4 | Execute$ TrigChooseRandom | TriggerZones$ Battlefield | TriggerDescription$ At the beginning of your upkeep, if there are four or more cards exiled with CARDNAME, choose one of them at random. You may cast it without paying its mana cost.
+SVar:TrigChooseRandom:DB$ ChooseCard | ChoiceZone$ Exile | AtRandom$ True | Choices$ Card.IsImprinted | SubAbility$ DBPlay 
+SVar:DBPlay:DB$ Play | Defined$ ChosenCard | WithoutManaCost$ True | SubAbility$ DBCleanup
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/river_songs_diary.txt:11]: ValidSA missing

File: forge-gui/res/cardsfolder/upcoming/the_curse_of_fenric.txt [hunk 4]
```diff
@@ -0,0 +1,15 @@
+Name:The Curse of Fenric
+ManaCost:2 G W
+Types:Enchantment Saga
+K:Saga:3:DBDestroy,DBAnimate,DBFight
+SVar:DBDestroy:DB$ Destroy | ValidTgts$ Creature | TargetMin$ 0 | TargetMax$ OneEach | TargetsWithDifferentControllers$ True | RememberLKI$ True | SubAbility$ DBToken | SpellDescription$ For each player, destroy up to one target creature that player controls. For each creature destroyed this way, its controller creates a 3/3 green Mutant creature token with deathtouch.
+SVar:DBToken:DB$ RepeatEach | UseImprinted$ True | DefinedCards$ DirectRemembered | RepeatSubAbility$ TokenMutant | ChangeZoneTable$ True | SubAbility$ DBCleanup
+SVar:TokenMutant:DB$ Token | TokenAmount$ 1 | TokenScript$ g_3_3_mutant_deathtouch | TokenOwner$ ImprintedController
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_curse_of_fenric.txt:7]: you don't need repeat for easy stuff

File: forge-gui/res/cardsfolder/upcoming/the_curse_of_fenric.txt [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
+SVar:DBAnimate:DB$ Animate | ValidTgts$ Creature.nonToken | TgtPrompt$ Select target nontoken creature | Power$ 6 | Toughness$ 6 | RemoveAllAbilities$ True | Types$ Legendary,Creature,Horror | Name$ Fenric | Duration$ Permanent | SpellDescription$ Target nontoken creature becomes a 6/6 legendary Horror creature named Fenric and loses all abilities.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_curse_of_fenric.txt:9]: need remove other creatue types

File: forge-gui/res/cardsfolder/upcoming/fugitive_of_the_judoon.txt (hunk 1 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/fugitive_of_the_judoon.txt:7]: @Simisays 
you need to move the first sentence into CostDesc, then it should look better
image

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/fugitive_of_the_judoon.txt:7]: A problem I've found is that you choose  to exile the human but not the artifact (meaning the effect will fizzle)

I think you need to choose them at the same time ? Is this possible currently ?

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/fugitive_of_the_judoon.txt:7]: nah that's a general problem for multiple cost parts
which might actually be a good reason to rewrite this with ChangeZone instead

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/fugitive_of_the_judoon.txt:7]: You mean two extra changezone lines (one for the artifact and the other for the human ?)

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/fugitive_of_the_judoon.txt]: isn't there something like EACH for that?

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/fugitive_of_the_judoon.txt]: I can't find an example where it's done together. I guess I'll just do it in two lines then

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/fugitive_of_the_judoon.txt]: no, it matters for triggers

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/fugitive_of_the_judoon.txt]: _Guided Passage_...

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/fugitive_of_the_judoon.txt]: Thanks this version seems to be working in my testing. Thoughts ?

File: forge-gui/res/cardsfolder/upcoming/river_songs_diary.txt [hunk 6]
```diff
@@ -0,0 +1,17 @@
+Name:River Song's Diary
+ManaCost:3
+Types:Artifact
+T:Mode$ SpellCast | ValidCard$ Instant,Sorcery | Execute$ TrigEffect | TriggerZones$ Battlefield | TriggerDescription$ Imprint — Whenever a player casts an instant or sorcery spell from their hand, exile it instead of putting it into a graveyard as it resolves.
+SVar:TrigEffect:DB$ Effect | ReplacementEffects$ ReMoved | RememberObjects$ TriggeredCard | SubAbility$ DBImprint
+SVar:DBImprint:DB$ Pump | ImprintCards$ TriggeredCard
+SVar:ReMoved:Event$ Moved | ValidCard$ Card.IsRemembered | Origin$ Stack | Destination$ Graveyard | Fizzle$ False | ReplaceWith$ DBExile | Description$ Exile it instead of putting it into a graveyard as it resolves.
+SVar:DBExile:DB$ ChangeZone | Defined$ ReplacedCard | Origin$ Stack | Destination$ Exile
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/river_songs_diary.txt:8]: shouldn't the effect clean itself up?

**Review by tool4ever — APPROVED** [review]
thanks

**Pull request merged**